lms/djangoapps/oauth2_handler/handlers.py

```diff
@@ -235,6 +240,6 @@ def _get_all_courses():
     """ Utility function to list all available courses. """
 
     ms_courses = modulestore().get_courses()
-    courses = [c for c in ms_courses if isinstance(c, CourseDescriptor)]
+    courses = [course for course in ms_courses if course.scope_ids.block_type == 'course']
```

Comment:
Yes. This was at the recommendation of @cpennington. See the outdated comment threads (to which I seem to be unable to link).